plugins/GSdx/GSTextureCacheSW.cpp

```diff
@@ -133,30 +121,26 @@ void GSTextureCacheSW::RemoveAll()
 
 void GSTextureCacheSW::IncAge()
 {
-	for(hash_set<Texture*>::iterator i = m_textures.begin(); i != m_textures.end(); )
+	for(auto i = m_textures.begin(); i != m_textures.end(); )
 	{
-		hash_set<Texture*>::iterator j = i++;
-
-		Texture* t = *j;
+		Texture* t = *i;
 
 		if(++t->m_age > 10)
 		{
-			m_textures.erase(j);
+			i = m_textures.erase(i);
```

Comment:
This code is only called once by frame so execution time is 0 even without optimization. Honestly micro-optimization is good for hot-path otherwise it is better to ensure readability of code.